core/src/main/java/org/pentaho/di/metastore/MetaStoreConst.java

```diff
@@ -0,0 +1,91 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************************/
+
+package org.pentaho.di.metastore;
+
+import java.io.File;
+
+import org.pentaho.di.core.Const;
+import org.pentaho.di.core.util.Utils;
+import org.pentaho.metastore.api.IMetaStore;
+import org.pentaho.metastore.api.exceptions.MetaStoreException;
+import org.pentaho.metastore.stores.xml.XmlMetaStore;
+import org.pentaho.metastore.stores.xml.XmlUtil;
+
+public class MetaStoreConst {
+
+  public static final String DB_ATTR_ID_DESCRIPTION = "description";
+  public static final String DB_ATTR_ID_PLUGIN_ID = "plugin_id";
+  public static final String DB_ATTR_ID_ACCESS_TYPE = "access_type";
+  public static final String DB_ATTR_ID_HOSTNAME = "host_name";
+  public static final String DB_ATTR_ID_PORT = "port";
+  public static final String DB_ATTR_ID_DATABASE_NAME = "database_name";
+  public static final String DB_ATTR_ID_USERNAME = "username";
+  @SuppressWarnings( "squid:S2068" ) public static final String DB_ATTR_ID_PASSWORD = "password";
+  public static final String DB_ATTR_ID_SERVERNAME = "server_name";
+  public static final String DB_ATTR_ID_DATA_TABLESPACE = "data_tablespace";
+  public static final String DB_ATTR_ID_INDEX_TABLESPACE = "index_tablespace";
+  public static boolean disableMetaStore; // Used for testing only
+
+  // Extra information for 3rd party tools, not used by Kettle
+  //
+  public static final String DB_ATTR_DRIVER_CLASS = "driver_class";
+  public static final String DB_ATTR_JDBC_URL = "jdbc_url";
+
+  public static final String DB_ATTR_ID_ATTRIBUTES = "attributes";
+
+  public static final String getDefaultPentahoMetaStoreLocation() {
+    return System.getProperty( "user.home" ) + File.separator + ".pentaho";
+  }
+
+  public static IMetaStore openLocalPentahoMetaStore() throws MetaStoreException {
+    return MetaStoreConst.openLocalPentahoMetaStore( true );
+  }
+
+  public static IMetaStore openLocalPentahoMetaStore( boolean allowCreate ) throws MetaStoreException {
+    if ( disableMetaStore ) {
+      return null;
+    }
+    String rootFolder = System.getProperty( Const.PENTAHO_METASTORE_FOLDER );
+    if ( Utils.isEmpty( rootFolder ) ) {
+      rootFolder = getDefaultPentahoMetaStoreLocation();
+    }
+    File rootFolderFile = new File( rootFolder );
+    File metaFolder = new File( rootFolder + File.separator + XmlUtil.META_FOLDER_NAME );
+    if ( !allowCreate && !metaFolder.exists() ) {
+      return null;
+    }
+    if ( !rootFolderFile.exists() ) {
+      try {
+        rootFolderFile.mkdirs();
+      } catch ( SecurityException e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)